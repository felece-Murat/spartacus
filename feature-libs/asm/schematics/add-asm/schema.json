{
  "$schema": "http://json-schema.org/schema",
  "$id": "AsmSchematics",
  "title": "ASM Schematics",
  "type": "object",
  "properties": {
    "project": {
      "type": "string",
      "description": "The name of the project.",
      "$default": {
        "$source": "projectName"
      }
    },
    "debug": {
      "description": "Display additional details during the running process.",
      "type": "boolean",
      "default": false
    },
    "lazy": {
      "type": "boolean",
      "description": "Lazy load the asm features.",
      "default": true
    },
    "features": {
      "type": "array",
      "uniqueItems": true,
      "default": ["ASM"]
    }
  },
  "required": []
}
