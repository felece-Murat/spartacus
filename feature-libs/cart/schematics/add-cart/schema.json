{
  "$schema": "http://json-schema.org/schema",
  "$id": "CartSchematics",
  "title": "Cart Schematics",
  "type": "object",
  "properties": {
    "project": {
      "type": "string",
      "description": "The name of the project.",
      "$default": {
        "$source": "projectName"
      }
    },
    "debug": {
      "description": "Display additional details during the running process.",
      "type": "boolean",
      "default": false
    },
    "lazy": {
      "type": "boolean",
      "description": "Lazy load the Saved cart feature.",
      "default": true
    },
    "features": {
      "type": "array",
      "uniqueItems": true,
      "items": {
        "enum": [
          "Cart",
          "WishList",
          "Saved-Cart",
          "Quick-Order",
          "Import-Export"
        ],
        "type": "string"
      },
      "default": [
        "Cart",
        "WishList",
        "Saved-Cart",
        "Quick-Order",
        "Import-Export"
      ],
      "x-prompt": {
        "message": "Which Cart features would you like to set up?",
        "type": "list",
        "items": [
          {
            "value": "Cart",
            "label": "Cart"
          },
          {
            "value": "WishList",
            "label": "Wish List"
          },
          {
            "value": "Saved-Cart",
            "label": "Saved Cart"
          },
          {
            "value": "Quick-Order",
            "label": "Quick Order"
          },
          {
            "value": "Import-Export",
            "label": "Import/Export"
          }
        ]
      }
    }
  },
  "required": []
}
