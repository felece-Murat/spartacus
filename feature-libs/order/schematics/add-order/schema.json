{
  "$schema": "http://json-schema.org/schema",
  "$id": "OrderSchematics",
  "title": "Order Schematics",
  "type": "object",
  "properties": {
    "project": {
      "type": "string",
      "description": "The name of the project.",
      "$default": {
        "$source": "projectName"
      }
    },
    "debug": {
      "description": "Display additional details during the running process.",
      "type": "boolean",
      "default": false
    },
    "lazy": {
      "type": "boolean",
      "description": "Lazy load the Order feature.",
      "default": true
    },
    "features": {
      "type": "array",
      "uniqueItems": true,
      "default": ["Order"]
    }
  },
  "required": []
}
