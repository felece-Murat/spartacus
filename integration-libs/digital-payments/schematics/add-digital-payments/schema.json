{
  "$schema": "http://json-schema.org/schema",
  "$id": "DigitalPaymentsSchematics",
  "title": "DigitalPayments Schematics",
  "type": "object",
  "properties": {
    "project": {
      "type": "string",
      "description": "The name of the project.",
      "$default": {
        "$source": "projectName"
      }
    },
    "debug": {
      "description": "Display additional details during the running process.",
      "type": "boolean",
      "default": false
    },
    "lazy": {
      "type": "boolean",
      "description": "Lazy load the digital payment.",
      "default": true
    },
    "features": {
      "type": "array",
      "uniqueItems": true,
      "default": ["Digital-Payments"]
    }
  },
  "required": []
}
