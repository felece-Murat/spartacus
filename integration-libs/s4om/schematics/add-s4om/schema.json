{
  "$schema": "http://json-schema.org/schema",
  "$id": "S4OMSchematics",
  "title": "S4OM Schematics",
  "type": "object",
  "properties": {
    "project": {
      "type": "string",
      "description": "The name of the project.",
      "$default": {
        "$source": "projectName"
      }
    },
    "debug": {
      "description": "Display additional details during the running process.",
      "type": "boolean",
      "default": false
    },
    "lazy": {
      "type": "boolean",
      "description": "Lazy load the S4OM.",
      "default": true
    },
    "features": {
      "type": "array",
      "uniqueItems": true,
      "default": ["S4HANA-Order-Management"]
    }
  },
  "required": []
}
