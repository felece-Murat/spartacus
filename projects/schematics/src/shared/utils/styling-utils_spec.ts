import { WorkspaceProject } from '@schematics/angular/utility/workspace-models';
import {
  getRelativeStyleConfigImportPath,
  getStylesConfigFilePath,
} from './styling-utils';

const sourceRoot = '/src';
const project = { sourceRoot } as WorkspaceProject;

describe.only('Styling utils', () => {
  describe('getStylesConfigFilePath', () => {
    it('should provide the path of the styles config file in the project source root.', () => {
      const stylesConfigFilePath = getStylesConfigFilePath(sourceRoot);
      expect(stylesConfigFilePath).toEqual(`${sourceRoot}/styles-config.scss`);
    });
  });

  describe('getRelativeStyleConfigImportPath', () => {
    it('should provide a correct relative path for a dest file in the same folder', () => {
      const relativeStyleConfigImportPath = getRelativeStyleConfigImportPath(
        project,
        '/src/styles.scss'
      );
      expect(relativeStyleConfigImportPath).toEqual(`styles-config`);
    });
    it('should provide a correct relative path for a dest file in a feature folder', () => {
      const relativeStyleConfigImportPath = getRelativeStyleConfigImportPath(
        project,
        '/src/spartacus/styles/feature.scss'
      );
      expect(relativeStyleConfigImportPath).toEqual(`../../styles-config`);
    });
  });
});
