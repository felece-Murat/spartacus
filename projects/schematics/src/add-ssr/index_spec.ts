import {
  SchematicTestRunner,
  UnitTestTree,
} from '@angular-devkit/schematics/testing';
import {
  Schema as ApplicationOptions,
  Style,
} from '@schematics/angular/application/schema';
import { Schema as WorkspaceOptions } from '@schematics/angular/workspace/schema';
import * as path from 'path';
import { Schema as SpartacusOptions } from '../add-spartacus/schema';
import { NGUNIVERSAL_EXPRESS_ENGINE } from '../shared/constants';
import { SPARTACUS_SCHEMATICS } from '../shared/libs-constants';
import { getPathResultsForFile } from '../shared/utils/file-utils';

const collectionPath = path.join(__dirname, '../collection.json');

describe('add-ssr', () => {
  const schematicRunner = new SchematicTestRunner(
    SPARTACUS_SCHEMATICS,
    collectionPath
  );

  let appTree: UnitTestTree;

  const workspaceOptions: WorkspaceOptions = {
    name: 'workspace',
    version: '0.5.0',
  };

  const appOptions: ApplicationOptions = {
    name: 'schematics-test',
    inlineStyle: false,
    inlineTemplate: false,
    routing: false,
    style: Style.Scss,
    skipTests: false,
    projectRoot: '',
  };

  const defaultOptions: SpartacusOptions = {
    project: 'schematics-test',
    baseSite: 'electronics',
    baseUrl: 'https://localhost:9002',
    lazy: true,
    features: [],
  };

  beforeEach(async () => {
    appTree = await schematicRunner
      .runExternalSchematicAsync(
        '@schematics/angular',
        'workspace',
        workspaceOptions
      )
      .toPromise();
    appTree = await schematicRunner
      .runExternalSchematicAsync(
        '@schematics/angular',
        'application',
        appOptions,
        appTree
      )
      .toPromise();
    appTree = await schematicRunner
      .runSchematicAsync(
        'add-spartacus',
        { ...defaultOptions, name: 'schematics-test' },
        appTree
      )
      .toPromise();
    appTree = await schematicRunner
      .runSchematicAsync(
        'add-ssr',
        { ...defaultOptions, name: 'schematics-test' },
        appTree
      )
      .toPromise();
  });

  describe('package.json', () => {
    it('should add SSR dependencies', async () => {
      const packageJson = appTree.readContent('/package.json');
      const packageObj = JSON.parse(packageJson);
      const depPackageList = Object.keys(packageObj.dependencies);

      expect(depPackageList.includes('@angular/platform-server')).toBe(true);
      expect(depPackageList.includes(NGUNIVERSAL_EXPRESS_ENGINE)).toBe(true);
      expect(depPackageList.includes('@spartacus/setup')).toBe(true);
    });

    it('should contain additional build scripts', async () => {
      const packageJson = appTree.readContent('package.json');

      const packageJsonFileObject = JSON.parse(packageJson);
      expect(packageJsonFileObject.scripts['build:ssr']).toBeTruthy();
      expect(packageJsonFileObject.scripts['serve:ssr']).toBeTruthy();
      expect(packageJsonFileObject.scripts['dev:ssr']).toBeTruthy();
    });
  });

  describe('server.ts', () => {
    it('should be configured properly', async () => {
      const serverTs = appTree.readContent('/server.ts');
      expect(serverTs).toMatchSnapshot();
    });
  });

  describe('app.server.module.ts', () => {
    it('should be updated', () => {
      const content = appTree.readContent('./src/app/app.server.module.ts');
      expect(content).toMatchSnapshot();
    });
  });

  describe('index.html', () => {
    it('should contain occ-backend-base-url attribute in meta tags', async () => {
      const indexHtmlPath = getPathResultsForFile(
        appTree,
        'index.html',
        '/src'
      )[0];
      const indexHtml = appTree.readContent(indexHtmlPath);
      expect(indexHtml).toMatchSnapshot();
    });
  });
});
