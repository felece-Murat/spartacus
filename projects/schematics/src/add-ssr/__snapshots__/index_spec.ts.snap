// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`add-ssr app.server.module.ts should be updated 1`] = `
"import { NgModule } from '@angular/core';
import { ServerModule } from '@angular/platform-server';

import { AppModule } from './app.module';
import { AppComponent } from './app.component';
import { provideServer } from '@spartacus/setup/ssr';

@NgModule({
  imports: [
    AppModule,
    ServerModule,
  ],
  bootstrap: [AppComponent],
  providers: [
    ...provideServer({
       serverRequestOrigin: process.env['SERVER_REQUEST_ORIGIN'],
     }),
  ],
})
export class AppServerModule {}
"
`;

exports[`add-ssr index.html should contain occ-backend-base-url attribute in meta tags 1`] = `
"<!doctype html>
<html lang="en">
<head>
  <meta charset="utf-8">
  <title>SchematicsTest</title>
  <base href="/">
  <meta name="viewport" content="width=device-width, initial-scale=1">
  <link rel="icon" type="image/x-icon" href="favicon.ico">
  <meta name="occ-backend-base-url" content="https://localhost:9002" />
</head>
<body>
  <app-root></app-root>
</body>
</html>
"
`;

exports[`add-ssr server.ts should be configured properly 1`] = `
"import 'zone.js/dist/zone-node';

import { ngExpressEngine as engine } from '@nguniversal/express-engine';
import { NgExpressEngineDecorator } from '@spartacus/setup/ssr';
import * as express from 'express';
import { join } from 'path';

import { AppServerModule } from './src/main.server';
import { APP_BASE_HREF } from '@angular/common';
import { existsSync } from 'fs';

const ngExpressEngine = NgExpressEngineDecorator.get(engine);

// The Express app is exported so that it can be used by serverless Functions.
export function app() {
  const server = express();
  const distFolder = join(process.cwd(), 'dist/schematics-test/browser');
  const indexHtml = existsSync(join(distFolder, 'index.original.html'))
    ? 'index.original.html'
    : 'index';

  server.set('trust proxy', 'loopback');

  server.engine(
    'html',
    ngExpressEngine({
      bootstrap: AppServerModule,
    })
  );

  server.set('view engine', 'html');
  server.set('views', distFolder);

  // Serve static files from /browser
  server.get(
    '*.*',
    express.static(distFolder, {
      maxAge: '1y',
    })
  );

  // All regular routes use the Universal engine
  server.get('*', (req, res) => {
    res.render(indexHtml, {
      req,
      providers: [{ provide: APP_BASE_HREF, useValue: req.baseUrl }],
    });
  });

  return server;
}

function run() {
  const port = process.env['PORT'] || 4000;

  // Start up the Node server
  const server = app();
  server.listen(port, () => {
    console.log(\`Node Express server listening on http://localhost:\${port}\`);
  });
}

// Webpack will replace 'require' with '__webpack_require__'
// '__non_webpack_require__' is a proxy to Node 'require'
// The below code is to ensure that the server is run only when not requiring the bundle.
declare const __non_webpack_require__: NodeRequire;
const mainModule = __non_webpack_require__.main;
const moduleFilename = (mainModule && mainModule.filename) || '';
if (moduleFilename === __filename || moduleFilename.includes('iisnode')) {
  run();
}

export * from './src/main.server';
"
`;
